applications/SwimmingDEMApplication/custom_utilities/qsvmsdemcoupled_data.h

```diff
@@ -0,0 +1,99 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Joaquin Gonzalez-Usua
+//
+
+#if !defined(KRATOS_QSVMSDEMCOUPLED_DATA_H)
+#define KRATOS_QSVMSDEMCOUPLED_DATA_H
+
+#include "../FluidDynamicsApplication/fluid_dynamics_application_variables.h"
+#include "../FluidDynamicsApplication/custom_utilities/fluid_element_data.h"
+#include "../FluidDynamicsApplication/custom_utilities/qsvms_data.h"
+#include "utilities/element_size_calculator.h"
+#include "includes/checks.h"
+
+namespace Kratos {
+
+///@addtogroup SwimmingDEMApplication
+///@{
+
+///@name Kratos classes
+///@{
+
+template< size_t TDim, size_t TNumNodes, bool TElementIntegratesInTime = false>
+class KRATOS_API(SWIMMING_DEM_APPLICATION) QSVMSDEMCoupledData : public QSVMSData<TDim, TNumNodes, TElementIntegratesInTime>
+{
+public:
+
+///@name Type Definitions
+///@{
+
+using NodalScalarData = typename FluidElementData<TDim,TNumNodes, false>::NodalScalarData;
+using NodalVectorData = typename FluidElementData<TDim,TNumNodes, false>::NodalVectorData;
+
+///@}
+///@name Public Members
+///@{
+
+NodalScalarData FluidFraction;
+NodalScalarData FluidFractionRate;
+
+NodalVectorData FluidFractionGradient;
+NodalVectorData Acceleration;
+NodalVectorData BodyForce;
+
+double ElementSize;
+
+///@}
+///@name Public Operations
+///@{
+
+void Initialize(const Element& rElement, const ProcessInfo& rProcessInfo) override
+{
+    // Base class Initialize manages constitutive law parameters
+    QSVMSData<TDim, TNumNodes, TElementIntegratesInTime>::Initialize(rElement,rProcessInfo);
+    const Geometry< Node<3> >& r_geometry = rElement.GetGeometry();
```

Comment:
Hey Ruben, could you please elaborate on this change? I think that your suggestion might be OK when writing code from scratch or when trying to replace long cumbersome iterators' names, but not for an already-existing code... I mean, why lose that extra piece of information?

Reply:
In my browser these lines are highlighted in green so I assumed that they are new. I agree that changing the type in already existent code difficults reviewing the new features (for this an independent PR with only refactoring changes must be created). The point of auto-deducing the type is that you are not tight to the type (2 classes with same API can be used) aside of the fact that minimizes the retrocompatibility effort (in this particular case, it would simplify a lot the migration once we finally remove the template from the node).

Reply:
Hi @rubenzorrilla , should we understand that using 'auto' is recommended over the classic declaration in general? I mean, it has obvious advantages, but also drawbacks (less readability, as it will take more time to know the type of a variable when trying to understand the code; less robustness during compilation, as it won't make any consistency check; ...). I guess this has been discussed within the Techincal Committee, but I am not aware of the official recommendation for Kratos.

Reply:
I think this is a case-by-case decision, I wouldn't give a general rule, esp as using `auto` also has some disadvantages

In this case I think it makes sense since as @rubenzorrilla says it will be easier to migrate if the template of the node is removed

Reply:
Thanks @philbucher , understood. Meanwhile I found this website for whoever is interested in the topic:
https://www.acodersjourney.com/c-11-auto/  describing several pros and cons.

Reply:
> Hi @rubenzorrilla , should we understand that using 'auto' is recommended over the classic declaration in general? I mean, it has obvious advantages, but also drawbacks (less readability, as it will take more time to know the type of a variable when trying to understand the code; less robustness during compilation, as it won't make any consistency check; ...). I guess this has been discussed within the Techincal Committee, but I am not aware of the official recommendation for Kratos.

About the readability, I tend to agree. However I'm not completely sure about the robustness of the compilation, if I'm not wrong it is deduced at compile time so I dare say it is as robust as using standard type declaration. In any case, it is true that you can misuse `auto` to intercept the return type, something that might lead to undefined behaviors (I remember facing this situation in the past). 

Having said this, I dare say that we should motivate (not promoting) the use of `auto` for all the cases in which the disadvantages are not apparent. For me the `GetGeometry()` is a good example.

Reply:
just to add that we currently have a situation where `auto` screws us up (even though it shouldn't) :)
https://github.com/KratosMultiphysics/Kratos/issues/7414

Reply:
Cool you pointed it out @philbucher . I've also experienced weird behavior from gcc with `auto` in the past. Sometimes it looks easy to us, but not so easy to gcc.